### DISCUSSION TITLE
Confused by Documentation term "token" used in Gateway Section in Step 3

### DISCUSSION THREAD

**Discussion by bmullan** [DefGuard/defguard#335] [Q&A] [answered]
On this page:
**[](https://defguard.gitbook.io/defguard/features/setting-up-your-instance/gateway)**

In STEP 3 it says - Fill in the default configuration file (/etc/defguard/gateway.toml)

When editing "/etc/defguard/gateway.toml" one entry asks for:

> \# **_Required: secret token generated by Defguard_**
> \# NOTE: must replace default with actual value
> **token = "<your_gateway_token>"**

Above, it says "_Required: **secret token generated by Defguard**_"

Where does this "secret token" get created & is that in the Documenation somwhere?
**-or-**
Is the "**_secret token_**" what is referred to elsewhere as one of these two??:
**DEFGUARD_GATEWAY_SECRET** : JWT secret key for encrypting Gateway tokens, default: DEFGUARD_GATEWAY_SECRET
**DEFGUARD_SECRET_KEY**: JWT secret key for encrypting private cookies; must be at least 64 characters long

Thanks

**Comment by teon** [comment, marked as answer]
When you launch the core/UI - after configuring your network there will be help how to setup the gateway and there also will be token for the gateway.
